### PR TITLE
Prepare for 22.0.0 release

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/5917 (split changelog into multiple files)

# Rationale for this change

- Prepare for 22.0.0 release
- Make changelog more manageable

# What changes are included in this PR?

- Update version to 22.0.0
- Split changelog into file per release because it was getting too large to work with
- [Rendered CHANGELOG](https://github.com/andygrove/arrow-datafusion/blob/prep-22/dev/changelog/22.0.0.md)

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/CHANGELOG.md [hunk 1]
```diff
@@ -19,3897 +19,24 @@
...
-- \[CI\] Fix the newly added linting errors to make clippy happy [\#3598](https://github.com/apache/arrow-datafusion/pull/3598) ([isidentical](https://github.com/isidentical))
-- Prevent over-allocations \(and spills\) on sorts with a fixed limit [\#3593](https://github.com/apache/arrow-datafusion/pull/3593) ([isidentical](https://github.com/isidentical))
-- update datafusion cli deps [\#3588](https://github.com/apache/arrow-datafusion/pull/3588) ([Jimexist](https://github.com/Jimexist))
-- Update cranelift\* dependencies `0.87` --\> `0.88` [\#3586](https://github.com/apache/arrow-datafusion/pull/3586) ([alamb](https://github.com/alamb))
-- Fix docs.rs [\#3580](https://github.com/apache/arrow-datafusion/pull/3580) ([avantgardnerio](https://github.com/avantgardnerio))
-- Fix build [\#3576](https://github.com/apache/arrow-datafusion/pull/3576) ([alamb](https://github.com/alamb))
-- Use consistent name for  `TimeUnit::Millisecond` [\#3575](https://github.com/apache/arrow-datafusion/pull/3575) ([alamb](https://github.com/alamb))
-- Fix logical plan serialization  [\#3574](https://github.com/apache/ar
```
Comment (alamb) [inline datafusion/CHANGELOG.md:22]: This is a nice change 👍

File: dev/changelog/22.0.0.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+## [22.0.0](https://github.com/apache/arrow-datafusion/tree/22.0.0) (2023-04-07)
+
+[Full Changelog](https://github.com/apache/arrow-datafusion/compare/21.1.0...22.0.0)
+
+**Breaking changes:**
+
+- Introduce a common trait TreeNode for ExecutionPlan, PhysicalExpr, LogicalExpr, LogicalPlan [#5630](https://github.com/apache/arrow-datafusion/pull/5630) (yahoNanJing)
+- Minor: Reduce clones in AnalyzerRule [#5728](https://github.com/apache/arrow-datafusion/pull/5728) (alamb)
+- Change required input ordering physical plan API to allow any NULLS FIRST / LAST and ASC / DESC [#5772](https://github.com/apache/arrow-datafusion/pull/5772) (mustafasrepo)
```
Comment (alamb) [inline dev/changelog/22.0.0.md:28]: I would like to try and get https://github.com/apache/arrow-datafusion/pull/5863 in (to minimize the external API churn). If it doesn't make ti that is ok

**Review by alamb — APPROVED** [review]
It would be great to fix https://github.com/apache/arrow-datafusion/issues/5756 as well but I don't think it is a release blocker

Thank you for keeping the 🚋  running

File: dev/changelog/22.0.0.md (hunk 2 above)
Comment (andygrove) [reply dev/changelog/22.0.0.md:28]: I am no rush, and generating the changelog is super fast these days

**Pull request merged**